Modules/System/Azure Blob Services API/src/ABSOperationResponse.Codeunit.al

```diff
@@ -61,6 +61,29 @@ codeunit 9050 "ABS Operation Response"
         Response := NewResponse;
     end;
 
+    procedure GetLeaseId(): Guid
```

Comment:
I'm not particularly fond of this approach.

So every set of functions would add their return value here? Can we do better?
Maybe a wrapping codeunit.. or an interface?

Reply:
Well, in my opinion the other way around this would be to only make the procedure `GetHeaderValueFromResponseHeaders` public, because there needs to be a way to retrieve values from the response headers.
A wrapping-codeunit would be a bit overkill (at least from what I think), because `"ABS Operation Response"` already is kind of a wrapping-codeunit for the HttpResponse.

But of course I'm open to suggestions 🤗

Reply:
I have a couple of different ideas:

_**Idea 1**_:
Have the lease ID as a return parameter in the function where it makes sense to be returned:
E.g
```
procedure LeaseAcquire(<some params>, var LeaseId: Guid): Codeunit "ABS Operation Response"
```

_**Idea 2**_:
It's a bit more work but hear me out...
Not necessarily wrap, but more like expose the "ABS Operation Response" via different interfaces.
E.g.
```
// Generic response
interface "ABS Operation Response"
{
    procedure IsSuccessful(): Boolean;
    procedure GetError(): Text;
}
```
```
// Response for lease-related operations
interface "ABS Lease Operation Response"
{
    // "inherited" from the generic one (here AL is still limiting us)
    procedure IsSuccessful(): Boolean;
    procedure GetError(): Text;
    
    // Lease-specific results
    procedure GetLeaseId(): Guid;
}
```

Then `LeaseAcquire` will look like
```
procedure LeaseAcquire(<some params>): Interface "ABS Lease Operation Response"
```

But `ListBlobs` will look like
```
procedure ListBlobs(<some params>): Interface "ABS Operation Response"
```

Now this approach requires codeunit "ABS Operation Response" to become _internal_ and to implement the interfaces so it can be accessed through them.


Ok, enough input from me. Let me hear your opinion!

Reply:
Wouldn't be **_Idea 2_** kind of a breaking change (making the existing public `"ABS Operation Response"` internal)? Of course, I wouldn't mind changing it like this. Would be similar like we have it in C#, e.g. with HttpResponseHeaders or HttpRequestHeaders, which inherits from HttpHeaders.

So, depending on if this is considered a breaking change I'd either proceed with Idea 1 or Idea 2

Reply:
Technically, yes. There's no way to directly mark it as internal.
The process is as usual — mark it as obsolete and, after X amount of time, mark it as internal.

I agree that the code would look ugly for a while.

Reply:
But then I'd propose to first add Idea 1 and then create another PR later introducing the refactoring (to not stretch this one out too much).
Would you agree?

Reply:
Changed it to Idea 1 first, to get this PR (hopefully 😉) done, as a first step.